docs/logging.md

```diff
@@ -0,0 +1,168 @@
+    _         <- loggingFS.warn("This is a WARN message")
+} yield ()
+```
+
+There are multiple ways to create `Logs` instance, among them:
+* `Logs.sync` that requires a `cats.effect.Sync` instance for delaying side-effects
+* `Logs.withContext` that requires a `cats.effect.Sync` instance for delaying side-effects and a proof that `F` has a context to log
+* `Logs.const` that wraps single `Logging` thus always returning the same instance
+* `Logs.empty` that will produce a no-op `Logging` instances
+* `Logs.combine` that combines two instances of `Logs`. 
+`Logging` instance that is created this way will be a combination of two underlying instances, produced by both `Logs`. 
+This means that both `Logging` implementations will be called in sequence    
+ 
+
+#### Logs representation (using LoggedValue and Loggable)
+User API in `Logging` typeclass showcases `LoggedValue` as a parameter to pass. 
+In fact, `LoggedValue` is a representation of a value that can be written to a log in a string form. 
+It's used internally, and you do not need to use it directly most of the time. Instead, what you want is a `Loggable` 
+typeclass, that describes how a value of some type `A` can be logged, both as a string representation in log message 
+and as a component of structured logging. Given an instance of `Loggable` for a type, a value of the type can be converted 
+into `LoggedValue` and thus logged in a way that you provided. 
+There are multiple predefined ways to create an instance of `Loggable`, many of them can be found in `tofu.logging.Loggable` object:  
+* `Loggable.empty` for no-op logging of value
+* `Loggable.show` for using `cats.Show` instance as string representaion
+* `Loggable.either` for logging either of `A` and `B`
+* provided instances for all primitive types, as well as stdlib's collections and collections from Cats
+* java.time.* instances
+
+Of course, you can describe your `Loggable` instance yourself by extending existing traits that pre-implement some functionality:
+* `DictLoggable` for multi-field objects
+* `ToStringLoggable` for using `.toString` for logging
+* `HideLoggable` to exclude value from logging
+
+#### Context logging
+It is possible to log the context of your computations (MDC, remember?) if you have an instance of `tofu.Context` for your
+effect type and a `Loggable` for your context. 
+That's natural - describe how we can obtain a context from a computation `F` and a way to represent that context in log, 
+and we will ensure it will be logged anytime you use `Logging` operations. 
+You can log that context to string as well as use it as a part of structured logging.
+
+#### Loggable example
+```scala mdoc
+import tofu.logging._
+
+case class User(name: String, surname: String)
+
+implicit val userLoggable = new DictLoggable[User] {
+  override def fields[I, V, R, S](a: User, i: I)(implicit r: LogRenderer[I, V, R, S]): R = {
+    r.addString("name", a.name, i)
```

Comment:
This is an incorrect implementation. 
It would work for mutable renderers only.
Any warning generator should warn you about illegal `R` dropping.
You mus use `|+|` combinator to compose multiple fields addition